### PR TITLE
Cmake config with dependencies

### PR DESCRIPTION
**Pull request by bytebunny**
At the moment, the users of the package need to manually import four additional dependencies before they can import `zq`. We can do it for them.

### CODE REVIEW

**Comment by a4z** [comment]
Thank you @bytebunny

This change requires additional changes, l
setting `FETCH_ASTR` in the top level CMakeList.txt  to default false, or remove it
I need to give it a moment to think about what way to go.   Think remove it the option, but then I also need to update the README :-)

File: cmake/zqConfig.cmake.in [hunk 1]
```diff
@@ -2,4 +2,10 @@
 
 include("${CMAKE_CURRENT_LIST_DIR}/zqTargets.cmake")
 
+include(CMakeFindDependencyMacro)
+find_dependency(astr)
+find_dependency(Protobuf)
```
Comment (a4z) [inline cmake/zqConfig.cmake.in]: do you have any suggestion on how that could work together with this PR
https://github.com/a4z/zq/pull/5
which makes Protobuf optional?

Comment (a4z) [reply cmake/zqConfig.cmake.in:7]: there are also other scenarios, like, when passing NO_CMAKE_FIND_ROOT_PATH to find package in some build configurations ....

Comment (bytebunny) [reply cmake/zqConfig.cmake.in]: We could add 
`set(Protobuf_DEPENDENCY "find_dependency(Protobuf)")` inside if statement https://github.com/a4z/zq/blob/d8de964359887c493f07bc746a76b39e79e00335/CMakeLists.txt#L76

and add `@Protobuf_DEPENDENCY@` to https://github.com/a4z/zq/blob/d8de964359887c493f07bc746a76b39e79e00335/cmake/zqConfig.cmake.in. 
That way, cmake will only look for Protobuf if the variable is set.

Comment (a4z) [reply cmake/zqConfig.cmake.in]: sounds like a good idea!

is it ok if I redirect the merge request to the feature/optional_proto branch so it goes together with https://github.com/a4z/zq/pull/5 as one change?

**Pull request merged**